From 0c7508f0a69591eb26d20c4a7cc1154884d75f0e Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Tue, 21 Aug 2012 16:22:18 +0800
Subject: [PATCH 30/36] Xilinx: ARM: EmacPs: Made changes to reflect actual
 hardware registers

This patch removes hash defines for some of the register bit
fields that does not exist anymore with the current version
of hardware. This also makes changes in the code in view of the same.

Signed-off-by: Anirudha <anirudh@xilinx.com>
Integrated-by: Hongbo Zhong <hongbo.zhong@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_emacps.c |    3 ++-
 drivers/net/ethernet/xilinx/xilinx_emacps.h |    3 ---
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_emacps.c b/drivers/net/ethernet/xilinx/xilinx_emacps.c
index be4b640..df871dc 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emacps.c
+++ b/drivers/net/ethernet/xilinx/xilinx_emacps.c
@@ -1497,7 +1497,8 @@ static void xemacps_init_hw(struct net_local *lp)
 	regval |= XEMACPS_NWCFG_FCSREM_MASK;
 	regval |= XEMACPS_NWCFG_PAUSEEN_MASK;
 	regval |= XEMACPS_NWCFG_100_MASK;
-	regval |= XEMACPS_NWCFG_1536RXEN_MASK;
+	regval |= XEMACPS_NWCFG_HDRXEN_MASK;
+
 #ifdef CONFIG_OF
 	if (lp->board_type == BOARD_TYPE_ZYNQ)
 		regval |= (MDC_DIV_224 << XEMACPS_NWCFG_MDC_SHIFT_MASK);
diff --git a/drivers/net/ethernet/xilinx/xilinx_emacps.h b/drivers/net/ethernet/xilinx/xilinx_emacps.h
index 908f370..e2bf733 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emacps.h
+++ b/drivers/net/ethernet/xilinx/xilinx_emacps.h
@@ -245,8 +245,6 @@ MDC_DIV_64, MDC_DIV_96, MDC_DIV_128, MDC_DIV_224 };
 #define XEMACPS_NWCFG_1000_MASK        0x00000400 /* Gigbit mode */
 #define XEMACPS_NWCFG_EXTADDRMATCHEN_MASK 0x00000200
 /* External address match enable */
-#define XEMACPS_NWCFG_1536RXEN_MASK    0x00000100 /* Enable 1536 byte
-						frames reception */
 #define XEMACPS_NWCFG_UCASTHASHEN_MASK 0x00000080 /* Receive unicast hash
 						frames */
 #define XEMACPS_NWCFG_MCASTHASHEN_MASK 0x00000040 /* Receive multicast hash
@@ -383,7 +381,6 @@ MDC_DIV_64, MDC_DIV_96, MDC_DIV_128, MDC_DIV_224 };
 #define XEMACPS_TXBUF_USED_MASK  0x80000000 /* Used bit. */
 #define XEMACPS_TXBUF_WRAP_MASK  0x40000000 /* Wrap bit, last descriptor */
 #define XEMACPS_TXBUF_RETRY_MASK 0x20000000 /* Retry limit exceeded */
-#define XEMACPS_TXBUF_URUN_MASK  0x10000000 /* Transmit underrun occurred */
 #define XEMACPS_TXBUF_EXH_MASK   0x08000000 /* Buffers exhausted */
 #define XEMACPS_TXBUF_LAC_MASK   0x04000000 /* Late collision. */
 #define XEMACPS_TXBUF_NOCRC_MASK 0x00010000 /* No CRC */
-- 
1.7.9.7

